workflow/scripts/enrichment_pathway_scatter.py

```diff
@@ -0,0 +1,184 @@
+        )
+        .mark_point(size=40)
+        .encode(
+            alt.Y(
+                effect_y,
+                title=effect_y,
+                scale=alt.Scale(type="log"),
+                axis=alt.Axis(grid=False),
+            ),
+            alt.X(
+                effect_x,
+                title=effect_x,
+                scale=alt.Scale(type="log"),
+                axis=alt.Axis(grid=False),
+            ),
+            color=alt.condition(
+                selector,
+                alt.Color(
+                    "order_color:N", scale=alt.Scale(range=color_scheme), legend=None
+                ),
+                alt.value("lightgray"),
+            ),
+            shape=alt.condition(
+                selector,
+                alt.Shape(
+                    "order_shape:N",
+                    scale=alt.Scale(
+                        domain=[0, 1, 2, 3, 4],
+                        range=[
+                            "triangle-up",
+                            "triangle-down",
+                            "square",
+                            "diamond",
+                            "cross",
+                        ],
+                    ),
+                    legend=None,
+                ),
+                alt.value("circle"),
+            ),
+            tooltip=[f"{name}:N", f"{effect_x}:Q", f"{effect_y}:Q"],
+        )
+        .add_params(selector)
+    )
+    return points
+
+
+def plot_legend(df_plot, name, selector, color_scheme):
+    legend = (
+        alt.Chart(df_plot)
+        .transform_calculate(
+            order_color=f"indexof({selector.name}.Name || [], datum.Name)",
+            order_shape=f"indexof({selector.name}.Name || [], datum.Name) % 5",
+        )
+        .mark_point(size=60)
+        .encode(
+            alt.Y(
+                name,
+                axis=alt.Axis(
+                    title="", ticks=False, domain=False, labelLimit=700, orient="right"
+                ),
+            ),
+            color=alt.condition(
+                selector,
+                alt.Color(
+                    "order_color:N", scale=alt.Scale(range=color_scheme), legend=None
+                ),
+                alt.value("lightgrey"),
+            ),
+            shape=alt.condition(
+                selector,
+                alt.Shape(
+                    "order_shape:N",
+                    scale=alt.Scale(
+                        domain=[0, 1, 2, 3, 4],
+                        range=[
+                            "triangle-up",
+                            "triangle-down",
+                            "square",
+                            "diamond",
+                            "cross",
+                        ],
+                    ),
+                    legend=None,
+                ),
+                alt.value("circle"),
+            ),
+        )
+        .transform_filter(selector)
+    )
+    return legend
+
+
+color_scheme = [
```

Comment:
Maybe add a comment, why you manually add a color scheme here and what the colors are?